redbot/cogs/streams/streams.py

```diff
@@ -773,10 +804,20 @@ async def _send_stream_alert(
         *,
         is_schedule: bool = False,
     ):
+        use_buttons: bool = await self.config.guild(channel.guild).use_buttons()
+        if use_buttons:
+            stream_url = embed.url
+            view = discord.ui.View()
+            view.add_item(
+                discord.ui.Button(
+                    label=_("Open Stream"), style=discord.ButtonStyle.link, url=stream_url
+                )
+            )
         m = await channel.send(
             content,
             embed=embed,
             allowed_mentions=discord.AllowedMentions(roles=True, everyone=True),
+            view=view if use_buttons else None,
```

Comment:
Simplification by removal of the ternary if operator and replacing it with always setting initial value of `view` to `None`.
```suggestion
        use_buttons: bool = await self.config.guild(channel.guild).use_buttons()
        view = None
        if use_buttons:
            stream_url = embed.url
            view = discord.ui.View()
            view.add_item(
                discord.ui.Button(
                    label=_("Open Stream"), style=discord.ButtonStyle.link, url=stream_url
                )
            )
        m = await channel.send(
            content,
            embed=embed,
            allowed_mentions=discord.AllowedMentions(roles=True, everyone=True),
            view=view,
```